collector/client.go

```diff
@@ -93,6 +94,49 @@ func (c *CacheObjectClient) GetCounters() (types.Counters, error) {
 	return counters, err
 }
 
+/*GetServiceTimes fetches service times from squid cache manager */
+func (c *CacheObjectClient) GetServiceTimes() (types.Counters, error) {
+	conn, err := connect(c.hostname, c.port)
+
+	if err != nil {
+		return types.Counters{}, err
+	}
+
+	r, err := get(conn, "service_times", c.basicAuthString)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if r.StatusCode != 200 {
+		return nil, fmt.Errorf("Non success code %d while fetching service_times metrics", r.StatusCode)
+	}
+
+	var serviceTimes types.Counters
+
+	reader := bufio.NewReader(r.Body)
+
+	for {
+		line, err := reader.ReadString('\n')
+
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		c, err := decodeServiceTimeStrings(line)
+		if err != nil {
+			log.Println(err)
+		} else {
+			serviceTimes = append(serviceTimes, c)
+		}
+	}
```

Comment:
nit: This part is also identical to the one above, extra bonus extract this and use a channel to read from the stream :)

Reply:
I gave it a crack :)